src/main/java/underdogs/devbie/favorite/service/FavoriteService.java

```diff
@@ -0,0 +1,24 @@
+package underdogs.devbie.favorite.service;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import underdogs.devbie.favorite.domain.Favorite;
+import underdogs.devbie.favorite.domain.FavoriteRepository;
+import underdogs.devbie.user.service.UserService;
+
+@Service
+@Transactional(readOnly = true)
+public abstract class FavoriteService<T extends Favorite> {
+
+    protected FavoriteRepository favoriteRepository;
+    protected UserService userService;
+
+    public abstract Object findFavorites(Long userId);
+
+    @Transactional
+    public abstract void createFavorite(Long objectId, Long userId);
+
+    @Transactional
+    public abstract void deleteFavorite(Long objectId, Long userId);
```

Comment:
```suggestion
    @Transactional
    public void deleteFavorite(Long objectId, Long userId) {
        내_즐겨찾기인지_확인();
        즐겨찾기_삭제();
    }
```
이런식으로 템플릿 메서드패턴을 가져가도 좋을 것 같습니다.

Reply:
삭제 메서드는 반환 타입이나 내부 로직이 똑같아서 템플릿 메서드 패턴을 이용해도 되겠네요.
이 부분은 템플릿 메서드 패턴을 적용해보도록 하겠습니다 :)